docs/data_quality.md

```diff
@@ -0,0 +1,111 @@
+### Valid
+
+As a prerequisite, it is assumed that a record passes [JSON Schema validation](#appendix-a-osv-schema-validation) for the version of the [OSV Schema](https://ossf.github.io/osv-schema/) it declares itself to comply with in the `schema_version` field, or 1.0.0 if it does not. It is also assumed that the vulnerability discussed in the OSV record is valid and affects the software described.
+
+### Precise
+
+A high quality OSV record allows a consumer of that record to be able to answer the following questions in an **automated** way, at scale:
+
+* "Does this vulnerability, as described, impact me?"
+  * "What version do I need to upgrade to, or what patches do I need to apply, for it not to impact me?"
+  * "Should I replace or remove this (potentially orphaned) package with known unfixed vulnerabilities?"
+
+The definition of “impact” will vary depending on how fine-grained the information available is (i.e. package-level or symbol-level for software library packages). Package-level precision is the minimum standard.
+
+* for version and commit ranges
+  * `affected[]`.`ranges[]`.`events[]`.`introduced` is defined
+  * prefer `affected[]`.`ranges[]`.`events[]`.`fixed` over `affected[]`.`ranges[]`.`events`.`last_affected`
+    * this minimizes false negatives
+  * distinct ranges for `introduced..fixed` and/or `introduced..last_affected` *(i.e. introduced and fixed versions or commits can't be the same)*
+  * values in `introduced` are before/less than `fixed`/`last_affected` according to the canonical package registry or repository
+  * for version (`ECOSYSTEM` and `SEMVER`) ranges
+    * the versions exist in the specific package ecosystem
+  * for commit (`GIT`) ranges
+    * the commits exist in the specified `repo` *(i.e. they are not from another GitHub fork)*
+* the `package.ecosystem`, and a unique `identifier` prefix for it, are defined in the OSV Schema
+* the `package.name` exists within the defined `package.ecosystem`, and is canonically encoded to be unambiguous *(i.e. normalized)*
+* Package URLs in the `package.url` field conform to the [specification](https://github.com/package-url/purl-spec)
+* `reference` URLs return a 2xx or 3xx response at the time of publication
+
+### Identifiable
+
+* Where relevant, an `alias` to the equivalent CVE record is present
+* Where an OSV record consolidates multiple vulnerabilities in another ecosystem (or universe), multiple `related` identifiers are present
+
+## Examples
+
+* [GO-2024-2687](https://api.osv.dev/v1/vulns/GO-2024-2687)
+  * Has `introduced` and `fixed` versions
+  * Has an alias to a CVE record ID
+  * Has a purl
+* [OSV-2024-98](https://api.osv.dev/v1/vulns/OSV-2024-98)
+  * Has `introduced` and `fixed` commits
+    * commits exist in repo
+* [DSA-5678-1](https://api.osv.dev/v1/vulns/DSA-5678-1)
+  * Has `introduced` and `fixed` versions
+  * Has multiple `related` CVE record IDs
+
+## Appendix A: OSV Schema validation
+
+(As at version 1.6.3, generated by Gemini)
```

Comment:
nit: let's say "generated by Gemini from the JSON schema" ?

Reply:
Addressed in bd9dcbfc